core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -703,7 +706,8 @@ public InodeDirectory getRoot() {
         newDir.setInternalAcl(pair.getFirst());
         newDir.setDefaultACL(pair.getSecond());
       }
-      mState.applyAndJournal(rpcContext, newDir);
+      mState.applyAndJournal(rpcContext, newDir,
+          inodePath.getUri().getLeadingPath(k));
```

Comment:
is there a way to avoid `getLeadingPath`? That method re-parses the path, which could be expensive.

Reply:
Refactored.